src/utils/ingestion_utils.py

```diff
@@ -23,12 +23,17 @@ def FetchSecret(secret_name):
     Example:
         secret_value = FetchSecret("mySecretName")
     """
-    credential = ManagedIdentityCredential(client_id=os.getenv("UAMI_CLIENT_ID"))
+
+    if os.getenv("AZURE_FUNCTIONS_ENVIRONMENT") == "Development":
+        credential =AzureCliCredential()
+    else:
+        credential = ManagedIdentityCredential(client_id=os.getenv("UAMI_CLIENT_ID")),  
     kv_url = os.getenv("KEY_VAULT_URI")
```

Comment:
Love the logic to select the appropriate credential based on the environment. Maybe later add logging to show which environment is being used currently.